baseapp/error.go

```diff
@@ -52,9 +58,16 @@ func HandleRouteError(w http.ResponseWriter, r *http.Request, err error) {
 	} else {
 		log = hlog.FromRequest(r).Error().Err(err)
 
-		WriteJSON(w, http.StatusInternalServerError, map[string]string{
-			"error": http.StatusText(http.StatusInternalServerError),
-		})
+		cause := errors.Cause(err)
+		if aerr, ok := cause.(HTTPError); ok {
+			http.Error(w, cause.Error(), aerr.StatusCode())
```

Comment:
I'm not sure which internal app you're updating, but I think it might be better to have `HandleRouteError` always produce a JSON response with a known format. If the error provides a message and status code, still render the JSON, but use the status and message from the error.